dbms/src/IO/DelimitedReadBuffer.h

```diff
@@ -36,7 +36,7 @@ class DelimitedReadBuffer : public ReadBuffer
                 return false;
 
             BufferBase::set(buffer->position(), buffer->available(), 0);
-            put_delimiter = true;
+            put_delimiter = (delimiter != 0);
```

Comment:
When using protobuf messages a single zero byte means a message with all its fields set to their default values. It seems there is no format where putting a single zero byte as a row delimiter could be useful. So I've decided not to put a row delimiter if it's set to `'\0'`.

Reply:
This change will break compatibility with previous scenarios where people may have used `\0` as delimiter.